File: iks-core/src/com/iks/cms/core/sql/query/UpdateColumn.java
package com.iks.cms.core.sql.query;

import com.iks.cms.core.sql.*;

public class UpdateColumn {
  private Column      column;
  private ColumnValue value;
  public UpdateColumn( Column column, Object value ) {
    this.column = column;
    this.value = new ColumnValue( value, column.getType() );
  }
  public Column getColumn() {
    return column;
  }
  public Object getValue() {
    return value.toString();
  }
  public String toString() {
    return column.getName() + "=" + value.toString();
  }
}

File: iks-core/src/com/iks/cms/core/sql/query/UpdateColumn.kt
package com.iks.cms.core.sql.query

import com.iks.cms.core.sql.*

class UpdateColumn(val column: Column, value: Any?) {
  var value: ColumnValue?

  init {
    this.value = ColumnValue(value, column.type)
  }

  override fun toString(): String {
    return column.name.quote() + "=" + value.toString()
  }
}